osu.Game/Overlays/Notifications/ProgressNotification.cs

```diff
@@ -17,49 +17,96 @@ namespace osu.Game.Overlays.Notifications
     public class ProgressNotification : Notification, IHasCompletionTarget
     {
         private string text;
+        public string Text
+        {
+            get { return text; }
+            set
+            {
+                text = value;
+                if (IsLoaded)
+                    textDrawable.Text = text;
+            }
+        }
 
         private float progress;
         public float Progress
         {
             get { return progress; }
             set
             {
-                Debug.Assert(state == ProgressNotificationState.Active);
                 progress = value;
-                progressBar.Progress = progress;
+                if (IsLoaded)
+                    progressBar.Progress = progress;
             }
         }
 
-        public ProgressNotificationState State
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            //we may have received changes before we were displayed.
+            State = state;
+            Progress = progress;

```

Comment:
That's a race condition though. If loading happens off the main thread IsLoaded won't be true within `load` and after Text has been set in there.